src/types.ts

```diff
@@ -180,6 +180,11 @@ export interface SDKCache {
 }
 
 export interface TxOverrides {
+  /**
+   * @deprecated we are going to calculate gas limit based on the transaction.
```

Comment:
I don't think this is deprecated.
`gasLimit` is still a valid optional parameter in `TxOverrides`, it just defaults to the vault from `getEstimatedGasLimit`
However, if the estimate fails and returns 0, we can still add `gasLimit` as a `txOverrides` parameter so it is valid.